### DISCUSSION TITLE
add the ability to read more environmet information. (list of compiler plugins,  target JVM version)

### DISCUSSION THREAD

**Discussion by rssh** [scala/scala3#18074] [Feature Requests]
Motivation - let us have a macro with functionality,  which have sense only if an appropriative compiler plugin is enabled.
Examples of use cases:
  -  async macro can call direct-style methods marked by context parameters only if the appropriate compiler plugin is in the same run.
  -  async macro can use the loom subsystem for transforming HO. method if target JVM is >= 21
  -  macro can generate places for inserting instrumentation calls by instrumentation plugin if one is enabled.

**Comment by ahoy-jon** [comment]
Interested too. I am currently writing a macro that would only make sense is Capture Checking is activated.
